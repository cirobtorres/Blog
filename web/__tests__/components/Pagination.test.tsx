import { render, screen } from "@testing-library/react";
import ArticlesPages from "../../src/components/Pagination";

describe("ArticlesPages", () => {
  it("renders the page it should be rendered, and does not render pages that should be hidden", () => {
    render(<ArticlesPages page={5} articlesCount={10} />);
    expect(screen.queryByText("1")).toBeInTheDocument();
    expect(screen.queryByText("2")).not.toBeInTheDocument();
    expect(screen.queryByText("3")).not.toBeInTheDocument();
    expect(screen.queryByText("4")).toBeInTheDocument();
    expect(screen.queryByText("5")).toBeInTheDocument();
    expect(screen.queryByText("6")).toBeInTheDocument();
    expect(screen.queryByText("7")).not.toBeInTheDocument();
    expect(screen.queryByText("8")).not.toBeInTheDocument();
    expect(screen.queryByText("9")).not.toBeInTheDocument();
    expect(screen.queryByText("10")).toBeInTheDocument();
  });

  it("renders two ellipses, one before and another after the current page", () => {
    render(<ArticlesPages page={5} articlesCount={10} />);

    const ellipses = screen.queryAllByText("More pages");
    expect(ellipses.length).toBe(2);

    const [firstEllipsis, secondEllipsis] = ellipses;
    const page4 = screen.getByText("4");
    const page6 = screen.getByText("6");

    expect(
      firstEllipsis.compareDocumentPosition(page4) &
        Node.DOCUMENT_POSITION_FOLLOWING
    ).toBeTruthy();
    expect(
      secondEllipsis.compareDocumentPosition(page6) &
        Node.DOCUMENT_POSITION_PRECEDING
    ).toBeTruthy();
  });

  it("renders aria-current=page to the corresponding page", () => {
    render(<ArticlesPages page={5} articlesCount={10} />);
    expect(screen.queryByText("1")).not.toHaveAttribute("aria-current", "page");
    expect(screen.queryByText("4")).not.toHaveAttribute("aria-current", "page");
    expect(screen.queryByText("5")).toHaveAttribute("aria-current", "page");
    expect(screen.queryByText("6")).not.toHaveAttribute("aria-current", "page");
    expect(screen.queryByText("10")).not.toHaveAttribute(
      "aria-current",
      "page"
    );
  });

  it("matches the snapshot", () => {
    const { asFragment } = render(
      <ArticlesPages page={5} articlesCount={10} />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
