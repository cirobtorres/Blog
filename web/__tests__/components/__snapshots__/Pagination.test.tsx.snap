// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticlesPages matches the snapshot 1`] = `
<DocumentFragment>
  <nav
    aria-label="pagination"
    class="mx-auto flex w-full justify-center"
    role="navigation"
  >
    <ul
      class="flex flex-row items-center gap-1"
    >
      <li
        class=""
      >
        <a
          class="inline-flex items-center justify-center gap-2 whitespace-nowrap rounded-md text-sm font-medium transition-all duration-500 focus-visible:outline-none focus-visible:outline-blog-foreground-readable-hover disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0 hover:bg-accent hover:text-accent-foreground h-9 w-9 max-[800px]:text-xs"
          href="?page=1"
        >
          1
        </a>
      </li>
      <li
        class=""
      >
        <span
          aria-hidden="true"
          class="flex h-9 w-9 items-center justify-center"
        >
          <svg
            class="lucide lucide-ellipsis size-4 max-[800px]:size-3"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <circle
              cx="12"
              cy="12"
              r="1"
            />
            <circle
              cx="19"
              cy="12"
              r="1"
            />
            <circle
              cx="5"
              cy="12"
              r="1"
            />
          </svg>
          <span
            class="sr-only"
          >
            More pages
          </span>
        </span>
      </li>
      <li
        class=""
      >
        <a
          class="inline-flex items-center justify-center gap-2 whitespace-nowrap rounded-md text-sm font-medium transition-all duration-500 focus-visible:outline-none focus-visible:outline-blog-foreground-readable-hover disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0 hover:bg-accent hover:text-accent-foreground size-9 max-[800px]:size-8 max-[800px]:text-xs"
          href="?page=4"
        >
          4
        </a>
      </li>
      <li
        class=""
      >
        <a
          aria-current="page"
          class="inline-flex items-center justify-center gap-2 whitespace-nowrap rounded-md text-sm font-medium transition-all duration-500 focus-visible:outline-none focus-visible:outline-blog-foreground-readable-hover disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0 text-blog-foreground-highlight bg-[hsl(0,0%,14.9%)] shadow-sm hover:bg-[hsl(0,0%,14.9%)] size-9 max-[800px]:size-8 max-[800px]:text-xs"
          href="?page=5"
        >
          5
        </a>
      </li>
      <li
        class=""
      >
        <a
          class="inline-flex items-center justify-center gap-2 whitespace-nowrap rounded-md text-sm font-medium transition-all duration-500 focus-visible:outline-none focus-visible:outline-blog-foreground-readable-hover disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0 hover:bg-accent hover:text-accent-foreground size-9 max-[800px]:size-8 max-[800px]:text-xs"
          href="?page=6"
        >
          6
        </a>
      </li>
      <li
        class=""
      >
        <span
          aria-hidden="true"
          class="flex h-9 w-9 items-center justify-center"
        >
          <svg
            class="lucide lucide-ellipsis size-4 max-[800px]:size-3"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <circle
              cx="12"
              cy="12"
              r="1"
            />
            <circle
              cx="19"
              cy="12"
              r="1"
            />
            <circle
              cx="5"
              cy="12"
              r="1"
            />
          </svg>
          <span
            class="sr-only"
          >
            More pages
          </span>
        </span>
      </li>
      <li
        class=""
      >
        <a
          class="inline-flex items-center justify-center gap-2 whitespace-nowrap rounded-md text-sm font-medium transition-all duration-500 focus-visible:outline-none focus-visible:outline-blog-foreground-readable-hover disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0 hover:bg-accent hover:text-accent-foreground size-9 max-[800px]:size-8 max-[800px]:text-xs"
          href="?page=10"
        >
          10
        </a>
      </li>
    </ul>
  </nav>
</DocumentFragment>
`;
